File: easy/1822_sign_of_the_product_of_an_array/solution.cpp
using namespace std;

#include <vector>

class Solution {
public:
  int arraySign(vector<int> &nums) {
    int numsProduct = 1;

    for (int &num : nums) {
      if (num == 0) {
        numsProduct = 0;
        break;
      } else if (num < 0) {
        numsProduct *= -1;
      }
    }

    if (numsProduct > 0)
      return 1;
    else if (numsProduct == 0)
      return 0;
    else
      return -1;
  }
};

File: easy/1822_sign_of_the_product_of_an_array/solution.py
from functools import reduce
from typing import List


class Solution:
    def arraySign(self, nums: List[int]) -> int:
        nums_product = reduce(lambda x, y: x * y, nums)

        if nums_product > 0:
            return 1
        elif nums_product == 0:
            return 0
        else:
            return -1
